bundles/AdminBundle/Resources/public/js/pimcore/object/tree.js

```diff
@@ -654,6 +654,27 @@ pimcore.object.tree = Class.create({
                 }
             }
 
+            // expand and collapse complete tree
+            if (record.data.type === "folder") {
```

Comment:
```suggestion
            if (!record.data.leaf) {
```
Also objects and variants can have children.